linker/files.go

```diff
@@ -271,6 +282,92 @@ func (r fileResolver) FindExtensionByNumber(message protoreflect.FullName, field
 	return r.deps.FindExtensionByNumber(message, field)
 }
 
+type aggResolver []Resolver
```

Comment:
This makes `ResolverFromFile` behave as documented. That doc states that the returned resolver uses the file's entire transitive closure. But it was actually only look at the file and its immediate imports. So this aggregate resolver is a recursive structure, so that resolution will actually search the entire closure (including files not directly imported, which also includes additional `deps` provided to the `Link` function, such as "google/protobuf/descriptor.proto" even if not imported).

Reply:
Never mind. I realized that the new behavior was not actually necessary to implement this PR. Furthermore (and a bit surprisingly), it had a small impact on the benchmarks (~1-3% slower, 2-3% more memory allocated). So I instead updated the docs to describe the actual behavior. (This repo is not yet v1.0, so this really isn't a breaking change and really is a doc fix.)